kernel/api/src/main/java/org/sakaiproject/serialization/MapperFactory.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2003-2017 The Apereo Foundation
+ *
+ * Licensed under the Educational Community License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *             http://opensource.org/licenses/ecl2
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sakaiproject.serialization;
+
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLOutputFactory;
+
+import com.ctc.wstx.api.WstxInputProperties;
+import com.ctc.wstx.api.WstxOutputProperties;
+import com.ctc.wstx.stax.WstxInputFactory;
+import com.ctc.wstx.stax.WstxOutputFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.dataformat.xml.XmlFactory;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import com.fasterxml.jackson.dataformat.xml.ser.ToXmlGenerator;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.fasterxml.jackson.datatype.jsr310.ser.InstantSerializer;
+
+public class MapperFactory {
```

Comment:
Just a note that ObjectMapper and XmlMapper are thread safe and can be reused

Reply:
note I think this class fine for setting up the mappers and configuring them though I think it would be better to follow the builder pattern

Reply:
```
ObjectMapper objectMapper = new MapperBuilder()
        .withSerializationFeature(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
        .withDeserializationFeature(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
        .json(); // or .xml();
```

Reply:
that is just some sample psuedo code yeah and I totally like the idea of updating BasicSerializationRepsoitory I would advocate for something similar for the Rubrics and Conversations Repositories

Reply:
I'm not sure I want to add toJSON and toXML methods to the SpringCrudRepository interface. Spring itself doesn't mix and match like that and I wanted to keep the SpringCrudRepository stuff reasonably similar to Spring's own to make it an easier switch to Spring proxied implementations of our interfaces. I like the mapper factory approach because it centralises that piece, standardises it across Sakai, but I'm not sure about going the whole hog and adding toXML and toJSON.

Reply:
How do you add serializer and deserializer properties to the mapper builder's factory though? I can't see how you configure the Woodstox stuff in the builder pattern.